src/inspector/main_thread_interface.cc

```diff
@@ -87,15 +87,17 @@ class CallRequest : public Request {
 
 class DispatchMessagesTask : public v8::Task {
  public:
-  explicit DispatchMessagesTask(MainThreadInterface* thread)
+  explicit DispatchMessagesTask(std::weak_ptr<MainThreadInterface> thread)
                                 : thread_(thread) {}
 
   void Run() override {
-    thread_->DispatchMessages();
+    std::shared_ptr<MainThreadInterface> thread = thread_.lock();
+    if (thread)
+      thread->DispatchMessages();
```

Comment:
This might be more idiomatic C++ as:
```c++
if (auto thread = thread_.lock()) thread->DispatchMessages();
```
But either way it's fine. (Ditto on line 244.)

Reply:
@bnoordhuis Thanks, done :)